plugins/logs/service_events

```diff
@@ -306,17 +348,13 @@ function fetch() {
         svcnm="$(echo "$service" | sed -r 's/^[^a-zA-Z]+//g' | sed -r 's/[^a-zA-Z0-9]+/_/g')"
         lognm="$(echo "$logfile" | sed -r 's/^[^a-zA-Z]+//g' | sed -r 's/[^a-zA-Z0-9]+/_/g')"
 
-        # Get previous line count to determine whether or not the file may have been rotated
+        # Get previous line count to determine whether or not the file may have been rotated (defaulting to 0)
         prvlines="$(echo "$curstate" | grep "^${lognm}_lines=" | cut -f 2 -d "=")"
-        if [ -z "$prvlines" ]; then
-            prvlines=0
-        fi
+        prvlines="${prvlines:-0}"
 
-        # Get the current number of lines in the file
+        # Get the current number of lines in the file (defaulting to 0 on error)
         curlines="$(wc -l < "$logfile")"
-        if ! [ "$curlines" -eq "$curlines" ] &>/dev/null; then
-            curlines=0
-        fi
+        curlines="${curlines:-0}"
```

Comment:
Is this conditional supposed to work around a missing `$logfile`?
Wouldn't the above assignment error out anyway in this case?

Reply:
Ah yes, it would err out, good call. Not sure what the best approach is in this case. Letting it err out and kill the script seems like the best answer, since the output to stderr should be sufficient for debugging purposes.

Reply:
Or better yet, put this conditional at the top of the loop, where it makes more sense.

Reply:
Thinking about it further: munin plugins are supposed to be fun to write - thus feel free to take a shortcut here, that *usually* works :)